base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -105,7 +105,9 @@ impl AggregateBody {
     }
 
     /// Should be used for tests only. Get a mutable reference to the inputs
+    /// This will set the sorted flag to false
     pub fn inputs_mut(&mut self) -> &mut Vec<TransactionInput> {
```

Comment:
These `*_mut` methods violate encapsulation and make maintaining an internal flag like `sorted` impossible. I would rather remove these methods and implement a better pattern

Reply:
yip agree there, its used in some integration tests, which means we cannot hide these behind `#cfg[test]`.